src/Settings/Settings.component.js

```diff
@@ -95,36 +99,63 @@ const SettingsDialog = React.createClass({
         const keys = _.get(this.tabs, key);
         const tabFields = _(fields).keyBy("name").at(keys).value();
 
-        return tabFields.map(field =>
-            FormHelpers.getSelectField({
-                name: field.name,
-                label: this.getTranslation(field.i18n_key),
-                isRequired: true,
-                options: field.options,
-                value: config[field.name],
-            })
-        );
+        return tabFields.map(field => {
+            if (field.options) {
+                return FormHelpers.getSelectField({
+                    name: field.name,
+                    label: this.getTranslation(field.i18n_key),
+                    isRequired: false,
+                    options: field.options,
+                    value: config[field.name],
+                });
+            } else {
+                return FormHelpers.getTextField({
+                    name: field.name,
+                    label: this.getTranslation(field.i18n_key),
+                    isRequired: true,
+                    value: config[field.name],
+                    validators: [{
+                        validator: Validators.isRequired,
+                        message: this.getTranslation(Validators.isRequired.message),
+                    }],
+                });
+            }
+        });
     },
 
     onChangeTab(value) {
         this.setState({currentTab: value});
     },
 
+    onUpdateFormStatus(section, status) {
+        const newFormStatuses = _(_.clone(this.state.formStatuses)).set(section, status.valid).value();
+        this.setState({formStatuses: newFormStatuses});
+    },
+
     render() {
-        const {loaded, fields, config} = this.state;
+        const {loaded, fields, config, formStatuses} = this.state;
+        const saveIsEnabled = loaded && _(formStatuses).values().every();
+
         const actions = [
             <FlatButton
                 label={this.getTranslation('cancel')}
                 onTouchTap={this.props.onRequestClose}
-                style={{ marginRight: 16 }}
+                style={{marginRight: 16}}
             />,
             loaded ? <RaisedButton
                 primary
                 label={this.getTranslation('save')}
+                disabled={!saveIsEnabled}
                 onTouchTap={this.save}
             /> : null,
         ];
 
+        const getTabCardProps = (section) => ({
+            fields: this.getFields(section),
+            onUpdateFormStatus: status => _.defer(this.onUpdateFormStatus, section, status),
+            onUpdateField: this.onUpdateField,
```

Comment:
@tokland Why are we adding these onUpdateFormStatus and onUpdateField? Out of curiosity...

Reply:
Those two are props of d2-ui:FormBuilder. `onUpdateField` usage is clear, `onUpdateFormStatus` on the other hand is call wthen the validity of the form changes. That way you can enable/disable save/next buttons or similar.